### DISCUSSION TITLE
Get data

### DISCUSSION THREAD

**Discussion by Ukhan11** [marcominerva/SearchGPT#2] [Q&A]
Hi there,

From where I need to upload my data?
can you please explain a bit about data.  also using this approach is there any prompts limits ?

any help on this will be really appreciated.

**Comment by marcominerva** [comment]
You need to manually create your Azure Cognitive Search index and then add its configuration to the appsettings.json file, as described in the README.md file.

For what concerns the prompt limits, it depends on how many documents you add to your index and how many results are returned by the Search index.

**Reply by Ukhan11** [reply]
Thank you for yours quick response.
Actually, my requirement is that I don't want to search for my data in multiple documents. I have one big file, and I want to send the large data, which is approximately 10k to 12k tokens, for the purpose of summarization.
So, is there any way to fulfill my requirements? If you have any knowledge on how we can send such longer prompts to GPT, it would be really helpful.

**Reply by marcominerva** [reply]
You can read the whole document and send its content as system message:

    var message = $$"""
        You are an assistant that knows the following information only:
        ---
        {{content}}
        ---
        You can use only the information above to answer questions. If you don't know the answer, reply suggesting to refine the question.
        """;
    
    var conversationId = await chatGptClient.SetupAsync(message);    

Where `content` is the actual content of your document. If you have 10k to 12k tokens, you need to use a model that supports an appropriate token numbers, like **gpt-3-5-16k** or **gpt-4-32k**.

**Reply by Ukhan11** [reply]
Hello,
I'm looking for the right Azure OpenAI API to fulfill my needs(mentioned earlier). Instead of using the GPT API, I prefer to use the Azure OpenAI. 
Could you provide some guidance on this?

Thanks in advance.

**Reply by marcominerva** [reply]
You can use Chat Completion API: https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions. These APIs are fully supported by **ChatGptNet**, the library on which this sample is built on: https://github.com/marcominerva/ChatGptNet.

**Reply by Ukhan11** [reply]
Thank you for the quick reply.

I have utilized both the Azure OpenAI Chat Completion API and the GPT API for the "gpt-3.5-turbo" model, but I am obtaining more accurate data from the GPT API. can you explain me about this.

prompt that I have used,
you will receive meeting notes, and your objective is to create a summary of the meeting that includes:

Overall summary of the discussion.
Action items, which entail what tasks need to be accomplished and who is responsible for them.
Pain points identified during the conversation.

**Reply by marcominerva** [reply]
It is difficult to say why a particular iteration of a model performs better than another. If you have a reproducible example, you can try to send them to Azure OpenAI support team and ask for clarification.
